### PR TITLE
Implemented ENV parametrization in YHS Web

### PR DESCRIPTION
**Pull request by dcoric**
This pull request includes several changes to the Makefile and web application configuration to support new environment variables and streamline the build process. The most important changes include modifications to the Makefile to add new environment variables, updates to the web application configuration files to support these variables, and improvements to the `EnvConfigService` class to handle the new URLs.

### Makefile Updates:
Modified the `web-build` target to include `yhsApiURL` and `yunikornApiURL` environment variables during the build process.

### Web Application Configuration:

* Added optional `yunikornApiURL` and `yhsApiURL` properties to the `EnvConfig` interface.
* Added `yunikornApiURL` and `yhsApiURL` properties to the environment configuration templates. 
* Updated the script to conditionally write to the production environment file.

### EnvConfigService Improvements:

* Updated the `EnvConfigService` class to initialize `yunikornApiURL` and `yhsApiURL` with default values and added methods to get these URLs.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -289,7 +298,9 @@ test-k6-performance: ## run k6 performance tests.
 
 .PHONY: web-build
 web-build: ng ## build the web components.
-	npm install --prefix web && npm run build --prefix web -- --base-href $(WEB_ROOT)
+	npm install --prefix web
+	yhsApiURL=http://127.0.0.1:8989 yunikornApiURL=$(strip $(call yunikorn_api_url)) npm run setenv --prefix web -- --base-href $(WEB_ROOT)
```
Comment (sudiptob2) [inline Makefile:302]: Should we hardcode `yhsApiURL` ? I think it can change based on the configuration in `config/yunikorn-history-server`

**Comment by dcoric** [comment]
> Can you please fix the build error?

Somehow I missed to update function names in the envconfig service. It is fixed now

**Review by nikola-jokic — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -61,7 +61,7 @@ GIT_TAG ?= $(shell git describe --tags --dirty --always)
 IMAGE_TAG ?= $(IMAGE_REPO):$(GIT_TAG)
 
 # WEB_ROOT defines path that will open web UI.
-WEB_ROOT ?= /web/
+WEB_ROOT ?= /
```
Comment (nikola-jokic) [inline Makefile:64]: Do we want to remove the web root? We can do it later if you think we can remove it since everything is served from `/` now

Comment (dcoric) [reply Makefile]: Yep, it is cleaner without it. I have removed it.

**Review by nikola-jokic — APPROVED** [review]

**Pull request merged**